### ISSUE TITLE
Regarding Login Failure in Guthub Desktop

### ISSUE THREAD

**Issue by Prakash-Balu** [shiftkey/desktop#947]
### The problem

Hi Team,

I have accessed my repository in Github Desktop but login authentication failed and retry and its not able to logged in Github desktop app in Ubuntu system. Kindly help me ASAP.
![Screenshot from 2023-10-19 23-57-57](https://github.com/shiftkey/desktop/assets/16771744/c3bce128-b9a2-44c7-b448-b406b26bebe0)

### Release version

Version 3.1.1-linux1 (x64)

### Operating system

Ubuntu x64

### Screenshots

![Screenshot from 2023-10-19 23-57-57](https://github.com/shiftkey/desktop/assets/16771744/1fc35fc4-fd8c-4ecc-b266-aabe190fabb5)

**Comment by S0litaire** [comment]
Same issue : 

OS: Ubuntu 23.10

Issue happens with both : 

GitHubDesktop-linux-amd64-3.3.3-linux2.deb
and
GitHubDesktop-linux-x86_64-3.3.3-linux2.AppImage

**Comment by shiftkey** [comment]
I need more details from the application to be able to troubleshoot further. Please check the logs created at `~/.config/GitHub\ Desktop/logs/` from around the time of the issue to see if there's any helpful messages that might explain the underlying behaviour.

**Comment by S0litaire** [comment]
Is their any flags for a more verbose output?

The only things showing in the logs is :

```
2023-10-24T23:30:50.448Z - info: [ui] [AppStore] loading 0 repositories from store
2023-10-24T23:30:50.483Z - info: [ui] launching: 3.3.3-linux2 (Linux 6.5.0-9-generic)
2023-10-24T23:30:50.483Z - info: [ui] execPath: '/usr/lib/github-desktop/github-desktop'
2023-10-24T23:30:55.269Z - info: [ui] [SignInStore] initializing OAuth flow
2023-10-24T23:30:55.273Z - info: [main] opening in browser: https://github.com/login/oauth/authorize?client_id=##########
```
Pressing the "cancel" button in the window does nothing, I have to click the "X" to close the window.

It opens my browser to the login authentication page and I log in.
Then it say's "You are being redirected to the authorized application"

![redirect](https://github.com/shiftkey/desktop/assets/1823781/067f8166-bb21-47d5-8edc-178ec7d1006d)

If I click on the "this setup page" link to open the "x-github-desktop-dev-auth://" link nothing happens either.


Additional information

Browser : 
Firefox 119.0 (64-bit) 
build: 119.0+build2-0ubuntu0.23.10.1~mt1   (Installed via Mozilla DEB packages)

In "about:preferences" The file association for "x-github-desktop-dev-auith" is set to "Use Github Desktop (default)" 
I've tried with the association set to "Github Desktop" and "to ask", their is no difference.

**Comment by shiftkey** [comment]
> If I click on the "this setup page" link to open the "x-github-desktop-dev-auth://" link nothing happens either.

Yeah, this suggests the browser isn't kicking in to handle the URL and launch the registerd app.

> In "about:preferences" The file association for "x-github-desktop-dev-auith" is set to "Use Github Desktop (default)"

I gather this is "x-github-desktop-dev-auth" (without the typo `i`) which is what the app registers so it's good to see that Firefox picks this up, but I'm puzzled about whether the Desktop app is successfully launching (or being re-launched with the right parameters).

One other thing to try - closing the app after it's stuck. This should flush any additional log messages to disk, which might indicate whether the app is being re-launched and for some reason not liking the parameters it's receiving.

**Comment by S0litaire** [comment]
Hi
Sorry for the typo it was 1am and i had no coffee! :)

Think I've gotten a **terrible** workaround to get it to work.

Their is something in the "apparmor" service which is blocking firefox from opening github-desktop during authentication

```
2023-10-25T13:24:51.680717+01:00 asus kernel: [   62.594220] audit: type=1400 audit(1698236691.676:188): apparmor="DENIED" operation="exec" class="file" profile="firefox" name="/usr/lib/github-desktop/github-desktop" pid=6054 comm="firefox" requested_mask="x" denied_mask="x" fsuid=1000 ouid=0
2023-10-25T13:24:54.004701+01:00 asus kernel: [   64.920039] audit: type=1400 audit(1698236694.000:189): apparmor="DENIED" operation="open" class="file" profile="firefox" name="/proc/5801/oom_score_adj" pid=5698 comm="firefox" requested_mask="w" denied_mask="w" fsuid=1000 ouid=1000
2023-10-25T13:24:54.008697+01:00 asus kernel: [   64.920281] audit: type=1400 audit(1698236694.004:190): apparmor="DENIED" operation="open" class="file" profile="firefox" name="/proc/5971/oom_score_adj" pid=5698 comm="firefox" requested_mask="w" denied_mask="w" fsuid=1000 ouid=1000
```

If I remove Firefox from the  "apparmor" service I can log in to github with no issues.


Used instructions from here to remove a process from apparmor:

https://ubuntu.com/server/docs/security-apparmor

>
>The /etc/apparmor.d/disable directory can be used along with the apparmor_parser -R option to disable a profile.
>```
>sudo ln -s /etc/apparmor.d/profile.name /etc/apparmor.d/disable/
>sudo apparmor_parser -R /etc/apparmor.d/profile.name
>```
>To re-enable a disabled profile remove the symbolic link to the profile in /etc/apparmor.d/disable/. Then load the profile using the -a option.
>```
>sudo rm /etc/apparmor.d/disable/profile.name
>cat /etc/apparmor.d/profile.name | sudo apparmor_parser -a
>```

**Comment by Jadeon-FPM** [comment]
I'm also having this issue. I had it over two years ago on the same account, and it's still ocurring after a fresh Ubuntu install. This is on an account that's only using repos that belong to an organization. It has never had issues on the Windows or Mac versions of GitHub Desktop. I also have a personal GitHub account that seems to work fine with this Linux version of Desktop.

I logged in to my GH account through the browser option, all seemed to work fine, the settings show me as logged in, and I can see a list of all repos my account has access to.

When I try to clone a repo I get the "Authentication failed" popup asking for credentials. If I clone a repo through CLI then add it to Desktop it will fail to push commits. Even if I try to log in at the auth failed popup with my username and a personal access token it still fails.

I have no issues cloning repos or pushing commits from CLI.

I also noticed that after installing GH Desktop it doesn't provide its own git binary in a way that's accessible from the terminal, and I still had to install git myself through apt. I don't know if it's supposed to or not.
